ext/storage_sources/azure_store/azure_connection.h

```diff
@@ -0,0 +1,24 @@
+#ifndef AZURECONNECTION
+#define AZURECONNECTION
+
+// Includes necessary for connection
+#include <azure/core.hpp>
+#include <azure/storage/blobs.hpp>
+
+#include <string>
+
+class azure_connection {
+    public:
+    azure_connection(const std::string &bucket_name, const std::string &obj_prefix = "");
+    int list_objects(std::vector<std::string> &objects) const;
+    int put_object(const std::string &file_name) const;
+    int delete_object() const;
+    int get_object(const std::string &path) const;
+    ~azure_connection() = default;
```

Comment:
No need for a default here, I would just not declare the destructor.

If the plan is to leave it here since we know stuff is going to go in it, then let's (a) move it up below the constructor and put a blank line after, and (b) mark it virtual. (The virtual thing is negotiable, I get that we're not planning to derive a class from this and there's a performance cost to having a vtable, but the kind of bugs you get from non-virtual destructors are awful to deal with.)